### DISCUSSION TITLE
License key

### DISCUSSION THREAD

**Discussion by clea-smudge** [skiptools/skip#58] [Q&A] [answered]
Hi, I already entered a license key I got from `https://skip.tools/eval/` into the `SKIPKEY: `field of the `~/.skiptools/skipkey.env` file. 

![Screenshot 2023-11-24 at 1 18 38 PM](https://github.com/skiptools/skip/assets/101157171/03625aab-80b1-468d-800b-919fb62d6673)

However I'm still getting this error when I try to build my Skip app. 

```No skip.tools license key found in ~/.skiptools/skipkey.env for transpilation of: {list of swift files} ```

I already tried
1. Clean and rebuild
2. Restarting XCode

Did I miss something? Do I need to add the key somewhere else?

Thanks!

**Comment by clea-smudge** [comment]
To bypass the error, for now, I just added this source code header comment in all of my Swift files manually

```
// This is free software: you can redistribute and/or modify it
// under the terms of the GNU Lesser General Public License 3.0
// as published by the Free Software Foundation https://fsf.org
```

and add a corresponding `LICENSE.LGPL` in the root folder of my project. Just curious though if there's any way or if there's a command that would put the above header in all my generated source files or can we only do that during project initialization?

**Reply by aabewhite** [reply, marked as answer]
Sorry for the trouble. Please make sure there is a space between the colon and the license key string. Also please make sure that the permissions on the file allow it to be read. 

Let us know if that doesn't work!

**Reply by GSK-FS** [reply]
asking space between colon and key worked for me
eg
SKIPKEY: Alldasldjasljd23jl

**Comment by X901** [comment]
> Sorry for the trouble. Please make sure there is a space between the colon and the license key string. Also please make sure that the permissions on the file allow it to be read.
> 
> Let us know if that doesn't work!

I have same issue
No skip.tools license key found in ~/.skiptools/skipkey.env for transpilation of

I add my licence, there is space between `#SKIPKEY:` and my key
but still show same issue
I did clean my project mutiple time !

is there something else I need to do ?

**Reply by marcprux** [reply]
Make sure you remove the "#" character; that makes it a comment (and thus is ignored).

If that doesn't help, can you post (or email to support@skip.tools) the contents of your `~/.skiptools/skipkey.env` file?

**Reply by X901** [reply]
Thank you, yup I forget to remove `#`
it working now
